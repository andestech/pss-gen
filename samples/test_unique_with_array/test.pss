component pss_top {

    action bit_a {
        rand bit [2:0] bit_val;
        
    }
    action root_a {
        rand bit [2:0] bit_arr[8];

        constraint {
            unique {bit_arr[0], bit_arr[1], bit_arr[2], bit_arr[3], bit_arr[4], bit_arr[5], bit_arr[6], bit_arr[7]};
        }

        exec body ASM  = """
	bit_arr[0] = {{bit_arr[0]}}
	bit_arr[1] = {{bit_arr[1]}}
	bit_arr[2] = {{bit_arr[2]}}
	bit_arr[3] = {{bit_arr[3]}}
	bit_arr[4] = {{bit_arr[4]}}
	bit_arr[5] = {{bit_arr[5]}}
	bit_arr[6] = {{bit_arr[6]}}
	bit_arr[7] = {{bit_arr[7]}}
        """;

    }

}

