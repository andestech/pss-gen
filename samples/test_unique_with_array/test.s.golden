
	bit_arr[0] = 2
	bit_arr[1] = 4
	bit_arr[2] = 7
	bit_arr[3] = 5
	bit_arr[4] = 6
	bit_arr[5] = 1
	bit_arr[6] = 3
	bit_arr[7] = 0
        
